apps/nxdiag: Update nxdiag app due to script place changes

eren-terzioglu opened the pull request:
## Summary

Related to https://github.com/apache/nuttx/pull/15304
Change path nxdiag script and remove system diagnostic scripts. Update suggested from https://github.com/apache/nuttx/pull/15304#discussion_r1894567114

## Impact

Common layer update

## Testing

`esp32c6-devkitc:nsh` config used with `SYSTEM_NXDIAG` option enabled

cederom reviewed (changes requested):
Thank you @eren-terzioglu :-)
* My question is why Python scripts are removed from `tools/` and how does that relate to `nxdiag` relocation?
* Why the removal takes place in this commit?
* There is no message explaining the change - what was changed into what and what changes are related?
* I guess you want to move some stuff from `nuttx/tools` to `nuttx-apps/../nxdiag`. But what if other folks used that `nuttx/tools` stuff already for other purpose? What if it has to be moved back again or re-created in `nuttx/tools` for some reason resulting in tools inconsistency?
* Couldn't `nxdiag` use the `nuttx/tools` stuff already as a most compatible way so we keep all tools in the `nuttx/tools`?
* Doesn't it require some documentation explanation?
* I am in general not fan of removing working stuff. If there is a bigger consensus for this please convince me otherwise :-)

eren-terzioglu commented:
> Thank you @eren-terzioglu :-)
> 
> * My question is why Python scripts are removed from `tools/` and how does that relate to `nxdiag` relocation?
> * Why the removal takes place in this commit?
> * There is no message explaining the change - what was changed into what and what changes are related?
> * I guess you want to move some stuff from `nuttx/tools` to `nuttx-apps/../nxdiag`. But what if other folks used that `nuttx/tools` stuff already for other purpose? What if it has to be moved back again or re-created in `nuttx/tools` for some reason resulting in tools inconsistency?
> * Couldn't `nxdiag` use the `nuttx/tools` stuff already as a most compatible way so we keep all tools in the `nuttx/tools`?
> * Doesn't it require some documentation explanation?
> * I am in general not fan of removing working stuff. If there is a bigger consensus for this please convince me otherwise :-)

- We moved `host_sysinfo.py` to `nuttx/tools` due to keep every script in one place for conveinence
- I changed the script for current nxdiag users, it is working without knowing anything. We can access `nuttx/tools` scripts (`host_sysinfo.py` in this case) through `apps/nxdiag` scripts
- Last but not least, suggestion came from [this comment](https://github.com/apache/nuttx/pull/15304#discussion_r1899563286). We can wait to see which implementation idea should be done.

cederom reviewed (approved):
Perfect! Thank you @eren-terzioglu !! :-)

I have restarted the CI jobs.. lets see if there are still some build errors.. probably comes from dependency on https://github.com/apache/nuttx/pull/15304 ? :-)

eren-terzioglu commented:
> Perfect! Thank you @eren-terzioglu !! :-)
> 
> I have restarted the CI jobs.. lets see if there are still some build errors.. probably comes from dependency on [apache/nuttx#15304](https://github.com/apache/nuttx/pull/15304) ? :-)

Thanks,
I updated and tested but seems error is still happening. I checked but there was no issue called `error`, `failed` or `warning` . I guess this lines reasons the error:

```
2025-01-07T08:45:11.1322592Z On branch master
2025-01-07T08:45:11.1323043Z Your branch is up to date with 'origin/master'.
2025-01-07T08:45:11.1323377Z 
2025-01-07T08:45:11.1323505Z Ignored files:
2025-01-07T08:45:11.1324175Z   (use "git add -f <file>..." to include in what will be committed)
2025-01-07T08:45:11.1324728Z 	tools/espressif/__pycache__/
```

Could it be?
